src/js/stores/preferences.js

```diff
@@ -141,17 +141,24 @@ define(function (require, exports, module) {
             _storage.setItem(PREF_INDEX, keysJSON);
             this.emit("change");
         },
-
+        
         /**
          * Set a single preference.
          *
          * @private
          * @param {{key: string, value: *}} payload
          */
-        _setPreference: function (payload) {
-            var key = payload.key,
-                value = payload.value;
+        _handleSetPreference: function (payload) {
+            this.setPreference(payload.key, payload.value);
+        },
 
+        /**
+         * Set a single preference.
+         *
+         * @param {sring} key
+         * @param {*} value
+         */
+        setPreference: function (key, value) {
```

Comment:
I open the access to the Preference#setPreference, so that we can avoid calling preference actions in the library store. Please let me know if there is a better way to achieve this.